docs/samples/Microsoft.ML.Samples/Dynamic/DataOperations/CrossValidationSplit.cs

```diff
@@ -0,0 +1,161 @@
+            var mlContext = new MLContext();
+
+            // Generate some data points.
+            var examples = GenerateRandomDataPoints(10);
+
+            // Convert the examples list to an IDataView object, which is consumable by ML.NET API.
+            var dataview = mlContext.Data.LoadFromEnumerable(examples);
+            // Cross validation splits your data randomly into set of "folds", and creates groups of Train and Test sets,
+            // where for each group, one fold is the Test and the rest of the folds the Train.
+            // So below, we specify Group column as the column containing the sampling keys.
+            // If we pass that column to cross validation it would be used to break data into certain chunks.
+            var folds = mlContext.Data.CrossValidationSplit(dataview, numberOfFolds: 3, samplingKeyColumnName: "Group");
+            PrintPreviewRows(folds[0]);
+
+            // The data in the Train split.
+            // [Group, 1], [Features, 0.8173254]
+            // [Group, 2], [Features, 0.7680227]
+            // [Group, 1], [Features, 0.2060332]
+            // [Group, 2], [Features, 0.5588848]
+            // [Group, 1], [Features, 0.4421779]
+            // [Group, 2], [Features, 0.9775497]
+            // 
+            // The data in the Test split.
+            // [Group, 0], [Features, 0.7262433]
+            // [Group, 0], [Features, 0.5581612]
+            // [Group, 0], [Features, 0.9060271]
+            // [Group, 0], [Features, 0.2737045]
+            PrintPreviewRows(folds[1]);
+            // The data in the Train split.
+            // [Group, 0], [Features, 0.7262433]
+            // [Group, 2], [Features, 0.7680227]
+            // [Group, 0], [Features, 0.5581612]
+            // [Group, 2], [Features, 0.5588848]
+            // [Group, 0], [Features, 0.9060271]
+            // [Group, 2], [Features, 0.9775497]
+            // [Group, 0], [Features, 0.2737045]
+            // 
+            // The data in the Test split.
+            // [Group, 1], [Features, 0.8173254]
+            // [Group, 1], [Features, 0.2060332]
+            // [Group, 1], [Features, 0.4421779]
+            PrintPreviewRows(folds[2]);
+            // The data in the Train split.
+            // [Group, 0], [Features, 0.7262433]
+            // [Group, 1], [Features, 0.8173254]
+            // [Group, 2], [Features, 0.7680227]
+            // [Group, 0], [Features, 0.5581612]
+            // [Group, 1], [Features, 0.2060332]
+            // [Group, 2], [Features, 0.5588848]
+            // [Group, 0], [Features, 0.9060271]
+            // [Group, 1], [Features, 0.4421779]
+            // [Group, 2], [Features, 0.9775497]
+            // [Group, 0], [Features, 0.2737045]
+            // 
+            // The data in the Test split.
+            // [Group, 2], [Features, 0.7680227]
+            // [Group, 2], [Features, 0.5588848]
+            // [Group, 2], [Features, 0.9775497]
```

Comment:
is it correct? It shows trainset has 10 example already???

Reply:
Good catch, let me debug this.

---
In reply to: [268364243](https://github.com/dotnet/machinelearning/pull/3058#discussion_r268364243) [](ancestors = 268364243)